File: VB/04_Shapes/HideShape.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation

Namespace HideShape
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load document from disk
			presentation.LoadFromFile("..\..\..\..\..\..\Data\FindShapeByAltText.pptx")

			'Loop through slides
			For Each slide As ISlide In presentation.Slides
				'Loop through shapes in the slide
				For Each shape As IShape In slide.Shapes
					'Find the shape whose alternative text is Shape1
					If shape.AlternativeText.CompareTo("Shape1") = 0 Then
						'Hide the shape
						shape.IsHidden = True
					End If
				Next shape
			Next slide

			'Save the document
			Dim result As String = "HideShape_result.pptx"
			presentation.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the file
			OutputViewer(result)
		End Sub
		Private Sub OutputViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/04_Shapes/HideShape.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;

namespace HideShape
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load document from disk
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\FindShapeByAltText.pptx");

            //Loop through slides
            foreach (ISlide slide in presentation.Slides)
            {
                //Loop through shapes in the slide
                foreach (IShape shape in slide.Shapes)
                {
                    //Find the shape whose alternative text is Shape1
                    if (shape.AlternativeText.CompareTo("Shape1") == 0)
                    {
                        //Hide the shape
                        shape.IsHidden = true;
                    }
                }
            }

            //Save the document
            string result = "HideShape_result.pptx";
            presentation.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the file
            OutputViewer(result);
        }
        private void OutputViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}